contracts/core/AuxiliaryCore.sol

```diff
@@ -47,6 +53,17 @@ contract AuxiliaryCore is AuxiliaryCoreInterface, AuxiliaryCoreConfig {
         bytes32 stateRoot;
     }
 
+    /**
+     * An auxiliary checkpoint.
+     *
+     * The blockhash is sufficient to identify the checkpoint. The height is
+     * required to know the epoch and enforce slashing conditions.
+     */
+    struct AuxiliaryCheckpoint {
```

Comment:
we want to know `gas` consumed, and report all auxiliary blocks.  We should report the RLP encoded block header, and decode it to obtain the parameters and hash it to obtain the block hash.  We can only report the child for a reported parent (extracted from header) so that we ensure an unbroken chain. When the reported block height is higher than `block.number - 256` we can assert that we are on the actual history.